modin/experimental/core/execution/native/implementations/hdk_on_native/__init__.py

```diff
@@ -12,3 +12,102 @@
 # governing permissions and limitations under the License.
 
 """Experimental Modin's functionality related to Native execution engine and optimized for HDK storage format."""
+
+import re
+import sys
+import modin.pandas as pd
+from modin.pandas.utils import from_arrow
+from modin.experimental.core.storage_formats.hdk import DFAlgQueryCompiler
+from modin.experimental.core.execution.native.implementations.hdk_on_native.hdk_worker import (
+    HdkWorker,
+)
+
+
+def hdk_query(query: str) -> pd.DataFrame:
+    """
+    Execute SQL queries on the HDK backend.
+
+    Data frames and local variables are referenced in the query
+    using the '@' prefix. All words in the query, prefixed with '@',
+    are replaced with the corresponding local or global variables.
+
+    Here is an example of a query to three data frames:
+
+    ids = [1, 2, 3]
+    first_names = ["James", "Peter", "Claus"]
+    last_names = ["Bond", "Pan", "Santa"]
+    courses_names = ["Mathematics", "Physics", "Geography"]
+    student = pd.DataFrame({"id": ids, "first_name": first_names, "last_name": last_names})
+    course = pd.DataFrame({"id": ids, "name": courses_names})
+    student_course = pd.DataFrame({"student_id": ids, "course_id": [3, 2, 1]})
+    sort_by = "last_name"
+    query = '''
+    SELECT
+        @student.first_name,
+        @student.last_name,
+        @course.name
+    FROM @student
+    JOIN @student_course
+    ON @student.id = @student_course.student_id
+    JOIN @course
+    ON @course.id = @student_course.course_id
+    ORDER BY
+        @sort_by
+    '''
+    res = hdk_query(query)
+    print(res)
+
+    Parameters
+    ----------
+    query : str
+        SQL query to be executed.
+
+    Returns
+    -------
+    modin.pandas.DataFrame
+        Execution result.
+    """
+    frame = sys._getframe()
+    try:
+        locals = frame.f_back.f_locals
+        globals = frame.f_back.f_globals
+    finally:
+        del frame
+
+    tmp_tables = {}
+    try:
+        off = 0
+        split = []
+        for match in re.finditer(r"@\w+", query):
+            start, end = match.span()
+            var_name = query[start + 1 : end]
+            if var_name in tmp_tables:
+                var_value = tmp_tables[var_name]
+            elif var_name in locals:
+                var_value = locals[var_name]
+            elif var_name in globals:
+                var_value = globals[var_name]
+            else:
+                raise KeyError(f"Neither local no global variable '{var_name}' found.")
+            split.append(query[off:start])
+
+            if isinstance(var_value, pd.DataFrame):
+                assert isinstance(var_value._query_compiler, DFAlgQueryCompiler)
+                mf = var_value._query_compiler._modin_frame
+                mf._filter_empties()
+                assert mf._has_arrow_table()
+                at = mf._partitions[0][0].get()
```

Comment:
```suggestion
                if not md._has_arrow_table():
                    mf._execute()
                at = mf._partitions[0][0].get()
                assert at is not None
```
maybe you intended logic something like that?

Reply:
Thanks, I'll check. If it works, I'll replace with this logic.